lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -396,7 +398,7 @@ def _configure_nodes(self, environment: Environment, log: Logger) -> None:
                 vm_name_prefix,
             )
 
-    def _configure_node(
+    def _configure_node(  # noqa: C901
```

Comment:
Because of lot of if/else in this function, it will thrown too complex function error which is C901 for flak8. There is no better way to handle this as those if/else are needed.

Reply:
In that case, we can refactor the new code to a new function for example `_configure_node_passthrough_devices` and call this fn from `_config_node`

Reply:
Still one if condition we have to add like below snippet, and this would cause the same error of C901

```
# Set device passthrough params for node context
        if node_runbook.device_passthrough:
            self._configure_device_passthrough(self, node_runbook, node_context)
```

Reply:
Does it still throw the same error if we move the condition to `_configure_device_passthrough` itself ?

Reply:
Since there would be condition in the function, there would be no harm in calling it everytime. I think it's worth giving it a try, atleast to see if this will eliminate format check error.

Reply:
That condition if function is to make mypy python 3.8 compliance or it was failing. I would not recommend to call the device passthrough config fn every time

Reply:
As I said earlier, if the following condition is inside the function, there is no harm in calling it every time. 
`if node_runbook.device_passthrough:`

`  # noqa: C901` has to the last option if nothing works. And I see the option I suggested still open.

Reply:
Ideally, we should skip the redundant call to function. Anyway, i made the change please check

Reply:
> There is no better way to handle this as those if/else are needed

If we make the function smaller by extracting bits out into functions, this error will go away, right?

Reply:
> > There is no better way to handle this as those if/else are needed
> 
> If we make the function smaller by extracting bits out into functions, this error will go away, right?

Even if i would put one if condition, it was coming. I wanted to call fn only when needed and avoid redundant call even when device pass through is not required.

I removed that condition also now. Please check the latest push

Reply:
Thanks for this change. Looks like we got rid of the flake8 error now.

Reply:
Yes that was because of IF we added. Now we removed it and calling fn every time